src/main/java/org/diego/tutorial/car/resources/CarResource.java

```diff
@@ -48,8 +60,22 @@ public Response getCars(@QueryParam("country") String country){
 					.build();
 	}
 	
+	/**
+	 * Method that adds a new car to the database.
+	 * @param car Car that needs to be added.
+	 * @return Car added
+	 */
 	@POST
 	public Response addCar(Car car) {
+		List<String> validationErrors = CarValidator.validateAddAndUpdate(car);
+		if (!validationErrors.isEmpty()) {
+			String message = "Request to add a car with non valid fields: ";
+			for (String validationError : validationErrors) {
+				message += validationError + " - ";
+			}
+			throw new BadRequestException(message);
+		}
+		

```

Comment:
this should be encapsulated into some method somewhere else

Reply:
Updated with encapsulation methods.